### PR TITLE
ENT-3277: Update travis to use stages.

### PR DESCRIPTION
**Pull request by Lex-2008**
Also provide a simple sample .travis.yml file

Note that this PR will break Travis building in other repos, so it should be merged together with relevant PRs to other repos, and after that all PRs in other repos should be rebased.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+sudo: required
+language: generic
+
+install: cd .. && git clone git@github.com:cfengine/buildscripts.git
```
Comment (jimis) [inline .travis.yml]: why is this needed?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      env: JOB=build
+      language: c
+      cache:
+        directories:
+          - $HOME/.cache/
+
+
+    - stage: test
+      env: JOB=acceptance-test
+      if: not type = cron
```
Comment (jimis) [inline .travis.yml]: why not run cron jobs?

File: build-scripts/travis [hunk 3]
```diff
@@ -1,85 +1,139 @@
...
+  echo $ARTIFACTS_KEY
+  echo $ARTIFACTS_SECRET
+  echo 
+  echo 
+  echo 
+  echo 
+  echo n
+  echo y
+) | s3cmd --configure >/dev/null
+
```
Comment (jimis) [inline build-scripts/travis]: lol

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    - stage: test
+      env: JOB=deployment-test
```
Comment (jimis) [inline .travis.yml]: +1 for parallelizing the different kinds of tests

File: build-scripts/travis [hunk 5]
```diff
@@ -1,85 +1,139 @@
...
+        s3cmd get $DEB_PACKAGE cfe.deb
+        sudo dpkg -i cfe.deb
+        touch /home/travis/rpmvercmp
+        chmod a+x ~/rpmvercmp
+        cd ..
+        test -d core || git clone git@github.com:cfengine/core.git
+        test -d enterprise || git clone git@github.com:cfengine/enterprise.git
+        chmod -R g-w */tests/acceptance
+        cd core/tests/acceptance
+        rm -rf ./05_processes/01_matching/timed/stime.cf ./05_processes/01_matching/promiser_match_is_correct.cf ./07_packages/00*.cf ./10_files/11_xml_edits ./25_cf-execd
```
Comment (jimis) [inline build-scripts/travis]: OK for now but create the relevant task to clean-up this.

File: build-scripts/travis [hunk 6]
```diff
@@ -1,85 +1,139 @@
...
+        AGENT=/var/cfengine/bin/cf-agent CF_PROMISES=/var/cfengine/bin/cf-promises CF_SERVERD=/var/cfengine/bin/cf-serverd CF_KEY=/var/cfengine/bin/cf-key CF_EXECD=/var/cfengine/bin/cf-execd CF_NET=/var/cfengine/bin/cf-net CF_RUNAGENT=/var/cfengine/bin/cf-runagent RPMVERCMP=~/rpmvercmp ./testall --tests=common,timed,slow,errorexit,serial,network,libxml2,libcurl,unsafe --gainroot=sudo
```
Comment (jimis) [inline build-scripts/travis]: Please break the long lines, it's very hard to read otherwise. 78 is the golden number, but feel free to deviate if needed. :-)

File: .travis.yml (hunk 1 above)
Comment (Lex-2008) [reply .travis.yml]: Probably this should be added as a comment on top:
```
# This is a sample file which should be put into other repos.
# Idea is that other repos (nova, enterprise, mission-portal) should check out
# this repo (buildscripts) and run common travis script from this repo
```
Indeed, this script is not expected to work with this repo.
Primarily, because it will fail on `git clone` step (`pwd` at the beginning of Travis run is something like /.../cfengine/buildscripts/, so after `cd ..` you're in a dir which already has a 'buildscripts' subdir)
Secondary, and that's more important, because buildscripts is a public repo, which doesn't have access to our private repos. This can be fixed, but not right now.

File: .travis.yml (hunk 2 above)
Comment (Lex-2008) [reply .travis.yml]: I'm thinking about using cron jobs only to refresh cache with dependency packages. Do you think we should run tests daily, too?

File: build-scripts/travis (hunk 5 above)
Comment (Lex-2008) [reply build-scripts/travis]: OK, https://tracker.mender.io/browse/CFE-2698 created!

File: .travis.yml (hunk 4 above)
Comment (Lex-2008) [reply .travis.yml]: Note, however, that CFEngine company is on the cheapest plan currently, so we don't really parallelize anything. Moreover, running two jobs sequentially in two different stages causes old VM to be destroyed and new one to be booted, thus taking time (about few minutes).

File: build-scripts/travis (hunk 6 above)
Comment (Lex-2008) [reply build-scripts/travis]: ok I've wrapped some lines and amended last commit

File: .travis.yml (hunk 2 above)
Comment (jimis) [reply .travis.yml]: Yes. Things some times cause breakage, which is detected after they are already in.

**Comment by jimis** [comment]
Thanks @Lex-2008, oneliners are now readable, feel like art.
